### PR TITLE
DashboardDataSource: Implement sharing logic inside the data source

### PR DESCRIPTION
**Pull request by torkelo**
In the scenes migration having as special data provider (ShareQueryDataProvider) turned out out to be a really bad idea as we really just want to treat it like any other panel (ie has a data source and queries). So treating it special created a lot of extra complexity. 

I first looked at ways of adding this functionality to SceneQueryRunner or via special runtime data source, but then realized we have a data source for it already!! 

Why is not the query logic handled inside DashboardDataSource?? I gave it a try and it just works and was super simple! 

* Handle looking up panel and subscribing to other panels data insde DashboardDataSource query method 
* Remove ShareQueryDataProvider 
* Handle showing DashboardDataSource QueryEditor like any other data source query editor (changed interface slightly) 
* Works for source panels that are not visible (not active)
* Works in panel edit (but needs a bit more work to get the panel edit time range)

Todo
* For Panel edit we still need copy the source panel into the edit scene so the source panel subscribes to the panel edit variable + and time range
* activation of source panel needs to be done inside the rxjs Observable so that the deactivation handler is called when the observable is completed.

### CODE REVIEW

File: public/app/features/dashboard-scene/panel-edit/PanelDataPane/PanelDataTransformations/PanelDataTransformationsTab.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import React from 'react';
```
Comment (torkelo) [inline public/app/features/dashboard-scene/panel-edit/PanelDataPane/PanelDataTransformations/PanelDataTransformationsTab.tsx:1]: oops, I must have based this PR on oscars PR

File: public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx [hunk 2]
```diff
@@ -154,6 +146,7 @@ export function buildPanelEditScene(dashboard: DashboardScene, panel: VizPanel):
         direction: 'column',
         primary: new SceneFlexLayout({
           direction: 'column',
+          minHeight: 200,
```
Comment (torkelo) [inline public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx:149]: PanelEdit css is a bit broken, when you open it with many queries there is no room for visualization (and you can't move the splitter, or scroll queries)

Comment (dprokop) [reply public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx:149]: Yes, @kaydelaney is already aware and will look into this :)

File: public/app/plugins/datasource/dashboard/datasource.ts [hunk 3]
```diff
@@ -20,8 +25,65 @@ export class DashboardDatasource extends DataSourceApi<DashboardQuery> {
...
+    }
+
+    if (!sourceDataProvider || !sourceDataProvider.getResultsStream) {
+      return of({ data: [] });
+    }
+
+    if (!sourceDataProvider?.isActive) {
+      sourceDataProvider?.activate();
+      sourceDataProvider.setContainerWidth!(500);
+    }
```
Comment (torkelo) [inline public/app/plugins/datasource/dashboard/datasource.ts:65]: this needs to be backed into the Observable so we can activate on subscriber start and unactivate when the stream is completed/closed. But we can do that in a follow-up PR

**Review by dprokop — APPROVED** [review]

**Pull request merged**